Sources/InteroperabilityTests/InteroperabilityTestCases.swift

```diff
@@ -217,6 +425,216 @@ struct ServerStreaming: InteroperabilityTest {
+///       }
+///       payload:{
+///         body: 27182 bytes of zeros
+///       }
+///     }
+///     ```
+///     If the call does not fail with `INVALID_ARGUMENT`, the test fails.
+///     Otherwise, we continue.
+///
+///  2. Client calls `StreamingInputCall` again, sending the *compressed* message
+///
+///     ```
+///     {
+///       expect_compressed:{
+///         value: true
+///       }
+///       payload:{
+///         body: 27182 bytes of zeros
+///       }
+///     }
+///     ```
+///
+///  3. And finally, the *uncompressed* message
+///     ```
+///     {
+///       expect_compressed:{
+///         value: false
+///       }
+///       payload:{
+///         body: 45904 bytes of zeros
+///       }
+///     }
+///     ```
+///
+///  4. Client half-closes
+///
+/// Client asserts:
+/// - First call fails with `INVALID_ARGUMENT`.
+/// - Next calls succeeds.
+/// - Response aggregated payload size is 73086.
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+class ClientCompressedStreaming: InteroperabilityTest {
+  func run(client: GRPCClient) async throws {
+    let testServiceClient = Grpc_Testing_TestService.Client(client: client)
+
+    // Does the server support this test? To find out we need to send an uncompressed probe. However
+    // we need to disable compression at the RPC level as we don't have access to whether the
+    // compression byte is set on messages. As such the corresponding code in the service
+    // implementation checks against the 'grpc-encoding' header as a best guess. Disabling
+    // compression here will stop that header from being sent.
+    let probeRequest: Grpc_Testing_StreamingInputCallRequest = .with {
+      $0.expectCompressed = .with { $0.value = true }
+      $0.payload = .with { $0.body = Data(repeating: 0, count: 27_182) }
+    }
+
+    var options = CallOptions.defaults
+    options.compression = CompressionAlgorithm.none
+
+    try await testServiceClient.streamingInputCall(
+      request: ClientRequest.Stream { writer in
+        try await writer.write(probeRequest)
+      },
+      options: options
+    ) { response in
+      switch response.accepted {
+      case .success:
+        throw AssertionFailure(message: "The result should be an error.")
+      case .failure(let error):
+        try assertEqual(error.code, .invalidArgument)
+      }
+    }
+
+    // Now for the actual test.
+
+    // The first message is identical to the probe message, we'll reuse that.
+    // The second should not be compressed.
+    let secondMessage: Grpc_Testing_StreamingInputCallRequest = .with { request in
+      request.expectCompressed = .with { $0.value = false }
+      request.payload = .with { $0.body = Data(repeating: 0, count: 45_904) }
+    }
+
+    options.compression = .gzip
+
+    try await testServiceClient.streamingInputCall(
+      request: ClientRequest.Stream { writer in
+        try await writer.write(probeRequest)
+        try await writer.write(secondMessage)
```

Comment:
We can't actually implement this test as we don't have control over the flags for individual messages, which means that in this case _both_ messages are compressed but the server is being told to have different expectations for each.